packages/opentelemetry-tracing/src/Span.ts

```diff
@@ -95,13 +95,25 @@ export class Span implements types.Span, ReadableSpan {
     return this;
   }
 
-  addEvent(name: string, attributes?: types.Attributes): this {
+  /**
+   *
+   * @param name Span Name
+   * @param attributes Span attributes
+   * @param startTime Specified start time for the event
+   */
+  addEvent(
+    name: string,
+    attributes?: types.Attributes,
+    startTime?: number
```

Comment:
This also allows passing in performance.now timestamps easily and more accurately as explained in above comment

Reply:
Does it make sense to change the order with a default value to startTime?

```
addEvent(name: string, 
         startTime:TimeInput = hrTime(), 
         attributes?: types.Attributes);
```

Reply:
I asked previously if this should be on hold until the time for event is added or I should add this here. My understanding was to add this here then and not to hold this PR longer.

Reply:
> Does it make sense to change the order with a default value to startTime?
> 
> ```
> addEvent(name: string, 
>          startTime:TimeInput = hrTime(), 
>          attributes?: types.Attributes);
> ```

1. `attributes` are already there, so it will break all possible places that this is being used as well as any of new code that is already being developed.
2. It will break the spec too
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#add-events
```
An Event is defined by the following properties:
(Required) Name of the event.
(Optional) One or more Attribute.
(Optional) Timestamp for the event.
```
time is on 3rd place.

3. Not sure what would be benefit besides great chance to break something in existing code

Reply:
My concern is it would be a problem if you give no `attributes` but a `startTime` to the function. Especially from JavaScript where you don't have the type. WDYT?

Reply:
Hmm not sure if I understand what you mean, can you give some example and what it may cause and how switching `startTime` with `attributes` will fix that ?

Reply:
With current signature, below are the valid scenarios:
```
span.addEvent('name'); // just a name
span.addEvent('name', {key: 'value'}); // name + attributes
span.addEvent('name', {key: 'value'}, Date.now()); // name + attributes + time
```

But, what if you have a `startTime` but no `attributes`?

`span.addEvent('name', Date.now()) // invalid params`
`span.addEvent('name', undefined, Date.now()) // valid but have to pass an undefined`

Is it a valid problem or no, what do you think?

Reply:
ok now it is clear, I can simply add detection if 2nd param is a valid type of `TimeInput` when arguments are 2 only and then treat it as time